plugin.video.vtm.go/resources/lib/modules/iptvmanager.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+""" Kodi PVR Integration module """
+
+from __future__ import absolute_import, division, unicode_literals
+
+import json
+import logging
+import socket
+from datetime import timedelta
+
+from resources.lib.modules import CHANNELS
+from resources.lib.vtmgo.vtmgo import VtmGo
+from resources.lib.vtmgo.vtmgoepg import VtmGoEpg
+
+_LOGGER = logging.getLogger('iptv-manager')
+
+
+class IPTVManager:
+    """ Code related to the Kodi PVR integration """
+
+    def __init__(self, kodi, port):
+        """ Initialise object
+        :type kodi: resources.lib.kodiwrapper.KodiWrapper
+        """
+        self._kodi = kodi
+        self._vtm_go = VtmGo(self._kodi)
+        self._vtm_go_epg = VtmGoEpg(self._kodi)
+        self.port = port
+
+    def via_socket(func):  # pylint: disable=no-self-argument
+        """Send the output of the wrapped function to socket"""
+
+        def send(self):
+            """Decorator to send over a socket"""
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
I'd say it's probably a better idea to place the connect call inside the call block and add an exception too in case this fails. But it is fine as is, it's not our responsibility to ask for these kind of changes.